### PR TITLE
Update group-avatar component

### LINKED ISSUE #16931: Group Avatar component doesn't have picture format
# Bug Report

## Problem 
[comment]: # (A problem description)
Current Group Avatar component implementation doesn't accept picture format

#### Expected behavior
https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=5245-133376&mode=design&t=FCCNvwubl2VhmrQh-4

#### Actual behavior
Only icon available now

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by codemaster115**
fixes #16931

### Summary

[comment]: # (Summarise the problem and how the pull request solves it)
The current group-avatar component implementation has some missing features.

- customization-color
- avatar

It will affect to group-avatar-tag component so need manual-qa for testing.

#### Platforms

- Android
- iOS
- macOS
- Linux
- Windows

status: ready

### CODE REVIEW

File: src/status_im2/contexts/quo_preview/avatars/group_avatar.cljs
Comment (ilmotta) [inline src/status_im2/contexts/quo_preview/avatars/group_avatar.cljs]: Our current guidelines suggest avoiding TODO comments in favor of just having issues. I see the code already had the comment before your PR, but it would be good if you could remove it since you are here. https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md#using-todos-comments

File: src/status_im2/contexts/quo_preview/avatars/group_avatar.cljs
Comment (ilmotta) [inline src/status_im2/contexts/quo_preview/avatars/group_avatar.cljs]: In Clojure we usually only use `boolean` if we strictly need something to be of the boolean type. In most conditional checks like in cond clauses this is unnecessary, so we can remove it here.

File: src/status_im2/contexts/quo_preview/avatars/group_avatar.cljs
Comment (ilmotta) [inline src/status_im2/contexts/quo_preview/avatars/group_avatar.cljs]: I guess this is a leftover from debugging the code?

File: src/status_im2/contexts/quo_preview/avatars/group_avatar.cljs
Comment (ilmotta) [inline src/status_im2/contexts/quo_preview/avatars/group_avatar.cljs]: https://github.com/status-im/status-mobile/blob/develop/doc/new-guidelines.md#always-add-styles-inside-the-style-key

File: src/quo2/components/avatars/group_avatar.cljs
Comment (ilmotta) [inline src/quo2/components/avatars/group_avatar.cljs]: The anonymous function overhead seems unnecessary because there's no initialization code outside the renderer.

File: src/status_im2/contexts/quo_preview/avatars/group_avatar.cljs
Comment (J-Son89) [reply src/status_im2/contexts/quo_preview/avatars/group_avatar.cljs]: Hmm, I think we should just try and fix this in this pr? 🤔 

In fact I think you are actually doing this here - so seems this comment is unnecessary?

File: src/quo2/components/avatars/group_avatar.cljs
Comment (J-Son89) [inline src/quo2/components/avatars/group_avatar.cljs]: perhaps we can move this to a view and style file while we are making the changes?

File: src/status_im2/contexts/quo_preview/avatars/group_avatar.cljs
Comment (codemaster115) [reply src/status_im2/contexts/quo_preview/avatars/group_avatar.cljs]: Yeah, it's unnecessary TODO

File: src/quo2/components/avatars/group_avatar.cljs
Comment (J-Son89) [inline src/quo2/components/avatars/group_avatar.cljs]: can we use `view-internal` and `view` to align with best practices?

File: src/quo2/components/avatars/group_avatar.cljs
Comment (J-Son89) [inline src/quo2/components/avatars/group_avatar.cljs]: btw, this changes the use of the component in the app so you need to update the uses to use key `:customization-color` in place of  `:color`

File: src/quo2/components/avatars/group_avatar.cljs
Comment (J-Son89) [inline src/quo2/components/avatars/group_avatar.cljs]: two things `avatar?` suggest a boolean because of the `?`. 

and similiarly in figma it is using `picture` so let's refer to it as that to keep things aligned 
Screenshot 2023-08-09 at 13 14 00

**Review by ilmotta — APPROVED** [review]
Can't speak for the UI, but the code LGTM :)

**Review by Francesca-G — APPROVED** [review]

**Comment by qoqobolo** [comment]
Hey @codemaster115, we noticed that this PR was also merged without QA approval, although the PR had the `request-manual-QA` label.
Could you please clarify if there were any specific reasons for this? I'd like to remind you that at the moment the process for PRs is as follows:
1. The PR must receive at least one approval from the dev review
2. Then the PR should be moved to the `e2e tests` column to run autotests (for all PRs)
3. If the developer is **_100% sure_** that changes in the PR cannot introduce new issues or regression, the `request-manual-QA` label can be omitted. Otherwise, the label should be added.
4. Then you should wait for manual testing from someone on the QA team. If issues are found, the QA will ping the developer in the comments and add the `tested issues` label.
If no issues are found, the QA will add the `tested-OK` label.

5. Then, depending on changes in the PR, this may also be sent for design review (corresponding column on the board).
If the designer finds issues (now Francesca is doing it), they add the `followup required` label.
After that, the PR is moved to the `Merge` column and can be merged.

6. If a design review is not required, the QA moves the PR to the merge column immediately.

You can also find the PR workflow instructions here (although it's a little outdated and doesn't include the new design review step).
https://github.com/status-im/status-mobile/blob/develop/doc/pipeline_process.md

Please feel free to contact us (in PRs or DM on Discord) with any questions!🙏

**Comment by codemaster115** [comment]
@qoqobolo Sorry for that. I just assumed this PR doesn't require QA since just changes the API of the component.
I will keep the process in the future. This PR is reverted now in https://github.com/status-im/status-mobile/pull/17039

**Pull request merged**
